From 7f0adaef68c5b0bb1c5eb9f5db5792b71b8b8beb Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Fri, 19 Mar 2010 10:44:09 +0100
Subject: [PATCH 6/6] Revert "ARM: SIMD: Try without any CFLAGS before forcing -mcpu="

This forces -marm that results in runtime SIGILL on thumb userspace

This reverts commit 18f0de452dc7e12e4cb544d761a626d5c6031663.
---
 configure.ac |   20 +++++---------------
 1 files changed, 5 insertions(+), 15 deletions(-)

diff --git a/configure.ac b/configure.ac
index fc3ee24..f84a4dc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -363,28 +363,18 @@ AM_CONDITIONAL(USE_VMX, test $have_vmx_intrinsics = yes)
 
 dnl ===========================================================================
 dnl Check for ARM SIMD instructions
-ARM_SIMD_CFLAGS=""
+ARM_SIMD_CFLAGS="-mcpu=arm1136j-s"
 
 have_arm_simd=no
 AC_MSG_CHECKING(whether to use ARM SIMD assembler)
-# check with default CFLAGS in case the toolchain turns on a sufficiently recent -mcpu=
+xserver_save_CFLAGS=$CFLAGS
+CFLAGS="$ARM_SIMD_CFLAGS $CFLAGS"
 AC_COMPILE_IFELSE([
 int main () {
     asm("uqadd8 r1, r1, r2");
     return 0;
-}], have_arm_simd=yes,
-    # check again with an explicit -mcpu= in case the toolchain defaults to an
-    # older one; note that uqadd8 isn't available in Thumb mode on arm1136j-s
-    # so we force ARM mode
-    ARM_SIMD_CFLAGS="-mcpu=arm1136j-s -marm"
-    xserver_save_CFLAGS=$CFLAGS
-    CFLAGS="$ARM_SIMD_CFLAGS $CFLAGS"
-    AC_COMPILE_IFELSE([
-    int main () {
-        asm("uqadd8 r1, r1, r2");
-        return 0;
-    }], have_arm_simd=yes)
-    CFLAGS=$xserver_save_CFLAGS)
+}], have_arm_simd=yes)
+CFLAGS=$xserver_save_CFLAGS
 
 AC_ARG_ENABLE(arm-simd,
    [AC_HELP_STRING([--disable-arm-simd],
-- 
1.6.6.1

